ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs

```diff
@@ -99,6 +103,8 @@ class
     -- Hard-fork related constraints
   , HasPartialConsensusConfig proto
   , FromCBOR (SL.PState (ProtoCrypto proto))
+  , Share (Core.TxOut era) ~ Interns (SL.Credential 'SL.Staking (ProtoCrypto proto))
```

Comment:
Where do these requirements come from? Seems unrelated to the rest of the diff.

Please continue the existing pattern by adding a comment that summarizes why they're needed and separates them from the "Hard-fork related constraints" above them.

Reply:
You're right.  I don't need them anymore.  Removed.